### PR TITLE
CODE RUB: AIFiles Exception Classes Update v2.10.1

### PR DESCRIPTION
**Pull request by Lezune**
Updated All Exceptions associated to the AIFiles directory to adhere to Anemic Data Models as well as updated the service layer to include the exception messaging.

### CODE REVIEW

File: Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/AIFileDependencyException.cs [hunk 1]
```diff
@@ -8,11 +8,6 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIFiles.Exceptions
 {
     public class AIFileDependencyException : Xeption
     {
-        public AIFileDependencyException(Xeption innerException)
-            : base(
-                message: "AI file dependency error occurred, contact support.",
-                    innerException: innerException)
-        { }
```
Comment (glhays) [inline Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/AIFileDependencyException.cs:11]: Remove extra line

File: Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/AIFileDependencyValidationException.cs [hunk 2]
```diff
@@ -8,11 +8,7 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIFiles.Exceptions
 {
     public class AIFileDependencyValidationException : Xeption
     {
-        public AIFileDependencyValidationException(Xeption innerException)
-            : base(
-                message: "AI file dependency validation error occurred, contact support.",
-                    innerException: innerException)
-        { }
+
```
Comment (glhays) [inline Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/AIFileDependencyValidationException.cs:11]: Extra line remove

File: Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/ExcessiveCallAIFileException.cs [hunk 3]
```diff
@@ -9,11 +9,6 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIFiles.Exceptions
 {
     public class ExcessiveCallAIFileException : Xeption
     {
-        public ExcessiveCallAIFileException(Exception innerException)
-            : base(
-                message: "Excessive call error occurred, limit your calls.",
-                    innerException: innerException)
-        { }
```
Comment (glhays) [inline Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/ExcessiveCallAIFileException.cs:12]: Extra line remove

File: Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/FailedAIFileServiceException.cs [hunk 4]
```diff
@@ -9,11 +9,6 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIFiles.Exceptions
 {
     public class FailedAIFileServiceException : Xeption
     {
-        public FailedAIFileServiceException(Exception innerException)
-            : base(
-                message: "Failed AI file service error occurred, contact support.",
-                    innerException: innerException)
-        { }
```
Comment (glhays) [inline Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/FailedAIFileServiceException.cs:12]: Same

File: Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/FailedServerAIFileException.cs [hunk 5]
```diff
@@ -9,11 +9,6 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIFiles.Exceptions
 {
     public class FailedServerAIFileException : Xeption
     {
-        public FailedServerAIFileException(Exception innerException)
-            : base(
-                message: "Failed AI file server error occurred, contact support.",
-                    innerException: innerException)
-        { }
```
Comment (glhays) [inline Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/FailedServerAIFileException.cs:12]: Same

File: Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/InvalidAIFileException.cs [hunk 6]
```diff
@@ -9,16 +9,10 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIFiles.Exceptions
...
-                message: "Invalid AI file error occurred, fix errors and try again.")
+        public InvalidAIFileException(string message)
+            : base(message)
         { }
 
-        public InvalidAIFileException(Exception innerException)
-            : base(
-                message: "Invalid AI file error occurred, fix errors and try again.",
-                    innerException: innerException)
-        { }
```
Comment (glhays) [inline Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/InvalidAIFileException.cs:16]: Extra line remove

File: Standard.AI.OpenAI/Models/Services/Foundations/AIModels/Exceptions/AIModelServiceException.cs [hunk 7]
```diff
@@ -8,11 +8,6 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIModels.Exceptions
 {
     public class AIModelServiceException : Xeption
     {
-        public AIModelServiceException(Xeption innerException)
-            : base(
-                message: "AI Model service error occurred, contact support.",
-                    innerException: innerException)
-        { }
```
Comment (glhays) [inline Standard.AI.OpenAI/Models/Services/Foundations/AIModels/Exceptions/AIModelServiceException.cs:11]: Extra line remove

File: Standard.AI.OpenAI/Models/Services/Foundations/AIModels/Exceptions/AIModelValidationException.cs [hunk 8]
```diff
@@ -8,11 +8,6 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIModels.Exceptions
 {
     public class AIModelValidationException : Xeption
     {
-        public AIModelValidationException(Xeption innerException)
-            : base(
-                message: "AI Model validation error occurred, fix errors and try again.",
-                    innerException: innerException)
-        { }
```
Comment (glhays) [inline Standard.AI.OpenAI/Models/Services/Foundations/AIModels/Exceptions/AIModelValidationException.cs:11]: same

File: Standard.AI.OpenAI.Tests.Unit/Services/Foundations/AIFiles/AIFileServiceTests.Validations.RemoveById.cs [hunk 9]
```diff
@@ -24,7 +24,7 @@ public partial class AIFileServiceTests
             string invalidFileId = invalidId;
 
             var invalidFileException =
-                new InvalidAIFileException();
+                new InvalidAIFileException(message: "Invalid AI file error occurred, fix errors and try again.");
```
Comment (glhays) [inline Standard.AI.OpenAI.Tests.Unit/Services/Foundations/AIFiles/AIFileServiceTests.Validations.RemoveById.cs:27]:
```suggestion
                new InvalidAIFileException(
                    message: "Invalid AI file error occurred, fix errors and try again.");
```

File: Standard.AI.OpenAI.Tests.Unit/Services/Foundations/AIFiles/AIFileServiceTests.Validations.Upload.cs [hunk 10]
```diff
@@ -52,7 +52,7 @@ private async Task ShouldThrowValidationExceptionOnUploadIfAIFileRequestIsNull()
             invalidAIFile.Request = null;
 
             var invalidAIFileException =
-                new InvalidAIFileException();
+                new InvalidAIFileException(message: "Invalid AI file error occurred, fix errors and try again.");
```
Comment (glhays) [inline Standard.AI.OpenAI.Tests.Unit/Services/Foundations/AIFiles/AIFileServiceTests.Validations.Upload.cs:56]:
```suggestion
               new InvalidAIFileException(
                message: "Invalid AI file error occurred, fix errors and try again.");
```

File: Standard.AI.OpenAI.Tests.Unit/Services/Foundations/AIFiles/AIFileServiceTests.Validations.Upload.cs [hunk 11]
```diff
@@ -99,7 +99,7 @@ private async Task ShouldThrowValidationExceptionOnUploadIfAIFileRequestIsInvali
             };
 
             var invalidAIFileException =
-                new InvalidAIFileException();
+                new InvalidAIFileException(message: "Invalid AI file error occurred, fix errors and try again.");
```
Comment (glhays) [inline Standard.AI.OpenAI.Tests.Unit/Services/Foundations/AIFiles/AIFileServiceTests.Validations.Upload.cs:102]:
```suggestion
                new InvalidAIFileException(
                    message: "Invalid AI file error occurred, fix errors and try again.");
```

**Comment by Lezune** [comment]
Hello @glhays Greg,

Thank you so much for the information as it is much appreciated. I will follow your advice and will study up on the best practices and guidelines to ensure I do not add unnecessary stress or work to the team. I will make sure I update the titles of my other pull requests after I reed the guidelines. Thanks again and I look forward to working with you.

File: Standard.AI.OpenAI.Tests.Unit/Services/Orchestrations/AIFiles/AIFileOrchestrationServiceTests.cs [hunk 12]
```diff
@@ -46,8 +46,8 @@ public static TheoryData DependencyValidationExceptions()
             {
                 new LocalFileValidationException(someInnerException),
                 new LocalFileDependencyValidationException(someInnerException),
-                new AIFileValidationException(someInnerException),
-                new AIFileDependencyValidationException(someInnerException)
+                new AIFileValidationException(message: "AI file validation error occurred, fix errors and try again.", someInnerException),
+                new AIFileDependencyValidationException(message: "AI file dependency validation error occurred, contact support.", someInnerException)
```
Comment (glhays) [inline Standard.AI.OpenAI.Tests.Unit/Services/Orchestrations/AIFiles/AIFileOrchestrationServiceTests.cs:50]: These lines exceed the 120 character line length limit, when they are broken then they need to be seperated by an extra line.

```suggestion

                new AIFileValidationException(
                    message: "AI file validation error occurred, fix errors and try again.",
                    innerException: someInnerException),
                
                new AIFileDependencyValidationException(
                    message: "AI file dependency validation error occurred, contact support.",
                    innerException: someInnerException)
```

File: Standard.AI.OpenAI.Tests.Unit/Services/Orchestrations/AIFiles/AIFileOrchestrationServiceTests.cs [hunk 13]
```diff
@@ -59,8 +59,8 @@ public static TheoryData DependencyExceptions()
             {
                 new LocalFileDependencyException(someInnerException),
                 new LocalFileServiceException(someInnerException),
-                new AIFileDependencyException(someInnerException),
-                new AIFileServiceException(someInnerException),
+                new AIFileDependencyException(message : "AI file dependency error occurred, contact support.", someInnerException),
```
Comment (glhays) [inline Standard.AI.OpenAI.Tests.Unit/Services/Orchestrations/AIFiles/AIFileOrchestrationServiceTests.cs:62]: Same here with these two lines

File: Standard.AI.OpenAI.Tests.Unit/Services/Orchestrations/AIFiles/AIFileOrchestrationServiceTests.cs [hunk 14]
```diff
@@ -70,8 +70,8 @@ public static TheoryData AIFileServiceDependencyExceptions()
 
             return new TheoryData<Xeption>
             {
-                new AIFileDependencyException(someInnerException),
-                new AIFileServiceException(someInnerException),
+                new AIFileDependencyException(message : "AI file dependency error occurred, contact support.", someInnerException),
```
Comment (glhays) [inline Standard.AI.OpenAI.Tests.Unit/Services/Orchestrations/AIFiles/AIFileOrchestrationServiceTests.cs:73]: Same Here

**Review by glhays — COMMENTED** [review]
@Lezune 
In your Service Classes Exceptions and Validations, these need refactored use the Service classes in ADotNet as an example. [ADotNet](https://github.com/The-Standard-Organization/ADotNet/blob/main/ADotNet/Services/Foundations/Files/FileService.Exceptions.cs)

**Comment by glhays** [comment]
@Lezune are you on the discord channel?

**Comment by Lezune** [comment]
@glhays I am on te discord channel. I changed my username to Lezune to make it easier to spot me. Also, thank you for the feedback.

File: Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/AIFileDependencyException.cs (hunk 1 above)
Comment (Lezune) [reply Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/AIFileDependencyException.cs]: This has been completed. Sorry for the wait.

File: Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/ExcessiveCallAIFileException.cs (hunk 3 above)
Comment (Lezune) [reply Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/ExcessiveCallAIFileException.cs]: This has been completed. Sorry for the wait.

File: Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/FailedAIFileServiceException.cs (hunk 4 above)
Comment (Lezune) [reply Standard.AI.OpenAI/Models/Services/Foundations/AIFiles/Exceptions/FailedAIFileServiceException.cs]: This has been completed. Sorry for the wait.

**Comment by glhays** [comment]
If my comments are fixed, please mark the comment resolved in the git IDE.

**Comment by Lezune** [comment]
Thank you for all of the help. All conversations have been marked as resolved.

**Comment by Lezune** [comment]
Updated and refactored code to reduce redundancy by adding methods in the AIFilesService class and calling the methods in the AIFilesService.Exception and AIFilesService.Validation classes as suggested by @glhays. 

Please ignore AIModel classes as they should be processes in the following Pull Requests:

#507 
#508 
#509 
#510 
#511 
#512 
#513 
#514 
#515 
#516 
#517

File: Standard.AI.OpenAI/Models/Services/Foundations/AIModels/Exceptions/NotFoundAIModelException.cs [hunk 15]
```diff
@@ -9,11 +9,6 @@ namespace Standard.AI.OpenAI.Models.Services.Foundations.AIModels.Exceptions
 {
     public class NotFoundAIModelException : Xeption
     {
-        public NotFoundAIModelException(Exception innerException)
-            : base(
-                message: "AI Model not found.",
-                    innerException: innerException)
-        { }
```
Comment (glhays) [inline Standard.AI.OpenAI/Models/Services/Foundations/AIModels/Exceptions/NotFoundAIModelException.cs:12]: Remove extra line

File: Standard.AI.OpenAI.Tests.Unit/Services/Foundations/AIFiles/AIFileServiceTests.Exceptions.RemoveById.cs [hunk 16]
```diff
@@ -25,13 +26,11 @@ private async Task ShouldThrowDependencyExceptionOnRemoveByIdIfUrlNotFoundAsync(
                 new HttpResponseUrlNotFoundException();
 
             var invalidConfigurationFileException =
-                new InvalidConfigurationAIFileException(
-                    message: "Invalid AI file configuration error occurred, contact support.",
+                createInvalidConfigurationAIFileException(
```
Comment (glhays) [inline Standard.AI.OpenAI.Tests.Unit/Services/Foundations/AIFiles/AIFileServiceTests.Exceptions.RemoveById.cs:29]: Remove all these in the unit test methods, as explained in Discord.
This comment refers to all, and will not repeat the comment for each instance there is a (create)

File: Standard.AI.OpenAI/Models/Services/Foundations/AIModels/Exceptions/NotFoundAIModelException.cs (hunk 15 above)
Comment (Lezune) [reply Standard.AI.OpenAI/Models/Services/Foundations/AIModels/Exceptions/NotFoundAIModelException.cs]: This has been resolved.

File: Standard.AI.OpenAI.Tests.Unit/Services/Foundations/AIFiles/AIFileServiceTests.Exceptions.RemoveById.cs (hunk 16 above)
Comment (Lezune) [reply Standard.AI.OpenAI.Tests.Unit/Services/Foundations/AIFiles/AIFileServiceTests.Exceptions.RemoveById.cs]: The Test unit files in the AIFiles directory were reverted to their original state. Had to update several lines as well as add another create method to resolve an error cause by a change in the Constructor of one of the updated Exceptions.

**Comment by glhays** [comment]
@Lezune I suggest that you do a similar merge with this PR, your not in an appropriate branch like
`user/lezune/coderub-aifiles-upgarde-v2101` but your going from from your forks main to main.
Don't worry about the comments you can always open the closed PR to review if needed.

**Comment by Lezune** [comment]
Thank you for the advice, @glhays. I will follow your suggestion and merge this into a new branch later today.

**Comment by Lezune** [comment]
Closing Pull Request. Resolved in PR #558.

**Pull request closed without merging**